### PR TITLE
A small piece of GAWB-2634: expose toProxyGroup() as a Sam endpoint

### PR DESCRIPTION
**Pull request by jmthibault79**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -398,6 +414,7 @@ paths:
           type: string
         - in: path
           description: Email of user to be added
+          name: email
```
Comment (jmthibault79) [inline src/main/resources/swagger/api-docs.yaml:417]: Did not pass the [Swagger validator](http://editor.swagger.io/#/) without these

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala [hunk 2]
```diff
@@ -22,7 +22,7 @@ import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
 class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: AccessPolicyDAO, val googleDirectoryDAO: GoogleDirectoryDAO, val googlePubSubDAO: GooglePubSubDAO, val googleIamDAO: GoogleIamDAO, val googleServicesConfig: GoogleServicesConfig, val petServiceAccountConfig: PetServiceAccountConfig)(implicit val executionContext: ExecutionContext) extends LazyLogging with FutureSupport with CloudExtensions {
-  private[google] def toProxyFromUser(subjectId: String): String = s"PROXY_$subjectId@${googleServicesConfig.appsDomain}"
+  private[google] def toProxyFromUser(subjectId: WorkbenchUserId): WorkbenchEmail = WorkbenchEmail(s"PROXY_${subjectId.value}@${googleServicesConfig.appsDomain}")
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala:25]: Updated signature for typesafety

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala [hunk 3]
```diff
@@ -85,6 +85,4 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
       deleteResult <- directoryDAO.deleteUser(userId)
     } yield deleteResult
   }
-
-  private[service] def toProxyFromUser(subjectId: String): String = s"PROXY_$subjectId@${cloudExtensions.emailDomain}"
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala:89]: Unused

**Comment by jmthibault79** [comment]
Swat test at https://github.com/broadinstitute/firecloud-ui/pull/1097

**Comment by rtitle** [comment]
LGTM

**Review by MatthewBemis — APPROVED** [review]
👍

**Review by helgridly — APPROVED** [review]

**Pull request merged**